FWCore/Utilities/src/TimeOfDay.cc

```diff
@@ -14,14 +15,18 @@ namespace edm {
 
   TimeOfDay::TimeOfDay(struct timeval const& tv) : tv_(tv) {}
 
+  TimeOfDay::TimeOfDay(std::chrono::microseconds const& duration)
+      : tv_{duration.count() / 1000000, duration.count() % 1000000} {}
+
   timeval TimeOfDay::setTime_() {
     timeval tv;
     gettimeofday(&tv, nullptr);
     return tv;
   }
 
   std::ostream& operator<<(std::ostream& os, TimeOfDay const& tod) {
-    std::ios::fmtflags oldflags = os.flags();  // Save stream formats so they can be left unchanged.
+    auto oldflags = os.flags();  // save the stream format flags so they can be restored
+    auto oldfill = os.fill();    // save the stream fill character so it can be restored
```

Comment:
looks like the fill character is not part of the flags, and has to be restored separately